smartsim/_core/launcher/step/dragonStep.py

```diff
@@ -90,3 +100,157 @@ def _get_exe_args_list(run_setting: DragonRunSettings) -> t.List[str]:
         exe_args = run_setting.exe_args
         args: t.List[str] = exe_args if isinstance(exe_args, list) else [exe_args]
         return args
+
+
+class DragonBatchStep(Step):
+    def __init__(
+        self,
+        name: str,
+        cwd: str,
+        batch_settings: t.Union[SbatchSettings, QsubBatchSettings],
+    ) -> None:
+        """Initialize a Slurm Sbatch step
+
+        :param name: name of the entity to launch
+        :type name: str
+        :param cwd: path to launch dir
+        :type cwd: str
+        :param batch_settings: batch settings for entity
+        :type batch_settings: SbatchSettings
+        """
+        super().__init__(name, cwd, batch_settings)
+        self.steps: t.List[Step] = []
+        self.managed = True
+        self.batch_settings = batch_settings
+        self._request_file_name = "requests.json"
+
+    def get_launch_cmd(self) -> t.List[str]:
+        """Get the launch command for the batch
+
+        :return: launch command for the batch
+        :rtype: list[str]
+        """
+        if isinstance(self.batch_settings, SbatchSettings):
+            script = self._write_sbatch_script()
```

Comment:
Can we re-use `SbatchStep` instead of duplicating code from that class in `_write_sbatch_script()`. 

Ditto on `_write_qsub_script`

Reply:
Maybe some settings copying mechanism that assumes user knows what's up, like:

```
BatchSettings:
  def transfer(self, source: BatchSettings) -> BatchSettings:
    # magic to copy internal state from source to self...

bs = SbatchSettings().transfer(self.batch_settings)
```

Reply:
I have tried this, but it seems to be just enough different to make it a bit hacky. I'll give it another attempt with a different approach, but I cannot assure I get anywhere with it.